Mark time_weight (and its rollup) parallel unsafe

syvb opened the pull request:
Parallel `time_weight` seems to cause issues when used with TimescaleDB compression.

(see https://github.com/timescale/Support-Dev-Collab/issues/3283)

syvb committed:
Mark time_weight (and its rollup) parallel unsafe

fabriziomello commented on extension/src/time_weighted_average.rs:
```diff
@@ -399,7 +399,7 @@ extension_sql!(
         serialfunc = time_weight_trans_serialize,\n\
         deserialfunc = time_weight_trans_deserialize,\n\
-        parallel = restricted\n\
+        parallel = unsafe\n\
```
I think you should also set `PARALLEL UNSAFE` for all internal functions:
* time_weight_trans
* time_weight_final
* time_weight_combine
* time_weight_trans_serialize
* time_weight_trans_deserialize

syvb replied:
I think the functions that make up the aggregates should still be parallel safe:

https://github.com/timescale/timescaledb-toolkit/blob/d895f628334769f852979df4a5ca6b2102487934/extension/src/time_weighted_average.rs#L173

antekresic committed:
Test that time_weight is not planned in parallel

Regression test for #794 / #878.

Forces the planner to consider parallel aggregation by setting
min_parallel_table_scan_size, parallel_setup_cost, parallel_tuple_cost
to 0 and max_parallel_workers_per_gather to 4 over a 50k-row table.
The EXPLAIN of "average(time_weight(...)) FROM tbl" must not contain
a Gather node; it would, if time_weight were ever again labelled
PARALLEL RESTRICTED/SAFE. The query is then run to ensure serial
execution succeeds.
